modules/payment/js/credit-card-validation.js

```diff
@@ -0,0 +1,258 @@
+/**
+ * @file
+ * Attaches credit card validation logic.
+ */
+
+(function ($) {
+
+  'use strict';
+
+  var types = {};
+  var VISA = 'visa';
+  var MASTERCARD = 'master-card';
+  var AMEX = 'amex';
+  var DINERSCLUB = 'dinersclub';
+  var DISCOVER = 'discover';
+  var MAESTRO = 'maestro';
+
+  types[VISA] = {
+    niceType: 'Visa',
```

Comment:
Why niceType and not label?
And why aren't the labels passed through Drupal.t()?